packages/scheduler/lib/models/tasks.ts

```diff
@@ -305,3 +306,36 @@ export async function expiresIfTimeout(db: knex.Knex): Promise<Result<Task[]>> {
         return Err(new Error(`Error expiring tasks: ${stringifyError(err)}`));
     }
 }
+
+export async function hardDeleteOlderThanNDays(db: knex.Knex, days: number): Promise<Result<Task[]>> {
+    try {
+        // Delete terminated tasks where lastStateTransitionAt is older than N days
+        // unless it is the most recent task for an given schedule
+        const tasks = await db
+            .from<DbTask>(TASKS_TABLE)
+            .where(
+                'id',
+                '=',
+                // schedules.id is null if the task is not associated with a schedule (ie: actions)
+                // or if the task is not the last task for its associated schedule
+                db.raw(`ANY(ARRAY(
+                    SELECT ${TASKS_TABLE}.id
+                    FROM ${TASKS_TABLE}
+                    LEFT JOIN ${SCHEDULES_TABLE} ON ${SCHEDULES_TABLE}.last_scheduled_task_id = ${TASKS_TABLE}.id
+                    WHERE ${TASKS_TABLE}.terminated = true
+                    AND ${TASKS_TABLE}.starts_after < NOW() - INTERVAL '${days} days'
```

Comment:
using `starts_after` because there is already an existing index for it. Didn't want to add another index and since `terminated = true` it doesn't matter which date we are using